src/main/java/com/yahoo/bullet/rest/service/QueryService.java

```diff
@@ -63,6 +69,24 @@ public void submit(String queryID, String query, QueryHandler queryHandler) {
         }
     }
 
+    /**
+     * Submit a signal query to Bullet.
+     *
+     * @param queryID The query ID to register request with.
+     * @param signal The {@link Signal} to be submitted.
+     */
+    public void submitSignal(String queryID, Signal signal) {
+        Publisher publisher = publisherRandomPool.get();
+        try {
+            Metadata metadata = new Metadata(signal, null);
+            PubSubMessage message = new PubSubMessage(queryID, null, metadata);
+            publisher.send(message);
+        } catch (Exception e) {
+            // Ignore failure.
```

Comment:
Is this safe? Or are we assuming that since all signals are KILL, the queryhandler will be closed anyway so no need to send anything back?

Reply:
Yes. queryHandler will be closed in line 87.